chainer/functions/normalization/batch_normalization.py

```diff
@@ -21,6 +21,7 @@ def check_type_forward(self, in_types):
         type_check.expect(
             x_type.dtype.kind == 'f',
             x_type.ndim >= gamma_type.ndim + 1,
+            x_type.shape[1] == gamma_type.shape[0],
```

Comment:
I think this condition supports only the case `ndim` of `gamma` equals to 1. If `x.shape` is `(B, s1, s2, ..., sN)`, then `gamma.shape` should be `(s1, ..., sM)` where `M <= N`. So the condition is `x_type.shape[1:1+M] == gamma_type.shape` where `M` is the `ndim` of `gamma`.